src/structures/BaseGuildTextChannel.js

```diff
@@ -0,0 +1,241 @@
+'use strict';
+
+const { Collection } = require('@discordjs/collection');
+const GuildChannel = require('./GuildChannel');
+const Webhook = require('./Webhook');
+const TextBasedChannel = require('./interfaces/TextBasedChannel');
+const MessageManager = require('../managers/MessageManager');
+const ThreadManager = require('../managers/ThreadManager');
+const DataResolver = require('../util/DataResolver');
+
+/**
+ * Represents a voice-based guild channel on Discord.
+ * @extends {GuildChannel}
+ * @implements {TextBasedChannel}
+ */
+class BaseGuildTextChannel extends GuildChannel {
+  /**
+   * @param {Guild} guild The guild the text channel is part of
+   * @param {APIChannel} data The data for the text channel
+   * @param {Client} [client] A safety parameter for the client that instantiated this
+   */
+  constructor(guild, data, client) {
+    super(guild, data, client);
```

Comment:
Similar to `GuildChannel` and `Channel` this probably needs to override the default behavior of patching immediately. Surprisingly it hasn't been an issue up to this point, but there is the potential for trying to access `this.messages` in _patch before it gets initialized